app/models/project_services/irker_service.rb

```diff
@@ -0,0 +1,149 @@
+# == Schema Information
+#
+# Table name: services
+#
+#  id         :integer          not null, primary key
+#  type       :string(255)
+#  title      :string(255)
+#  project_id :integer          not null
+#  created_at :datetime
+#  updated_at :datetime
+#  active     :boolean          default(FALSE), not null
+#  properties :text
+#
+
+require 'uri'
+require 'socket'
+
+class IrkerService < Service
+  prop_accessor :colorize_messages, :recipients, :channels
+  validates :recipients, presence: true, if: :activated?
+  validate :check_recipients_count, if: :activated?
+
+  before_validation :get_channels
+
+  def title
+    'Irker (IRC gateway)'
+  end
+
+  def description
+    'Send IRC messages, on update, to a list of recipients through an Irker '\
+    'gateway.'
+  end
+
+  def help
+    msg = 'Recipients have to be specified with a full URI: '\
+    'irc[s]://irc.network.net[:port]/#channel. Special cases: if you want '\
+    'the channel to be a nickname instead, append ",isnick" to the channel '\
+    'name; if the channel is protected by a secret password, append '\
+    '"?key=secretpassword" to the URI.'
+
+    begin
+      default_irc = Gitlab.config.irker['default_irc_uri']
+    rescue Settingslogic::MissingSetting
+    end
+    unless default_irc.nil?
+      msg += ' Note that a default IRC URI is provided by this service\'s '\
+      "administrator: #{default_irc}. You can thus just give a channel name."
+    end
+    msg
+  end
+
+  def to_param
+    'irker'
+  end
+
+  def execute(push_data)
+    IrkerWorker.perform_async(project_id, channels,
+                              colorize_messages, push_data)
+  end
+
+  def fields
+    [
+      { type: 'textarea', name: 'recipients',
+        placeholder: 'Recipients/channels separated by whitespaces' },
+      { type: 'checkbox', name: 'colorize_messages' },
+    ]
+  end
+
+  private
+
+  def check_recipients_count
+    return true if recipients.nil? || recipients.empty?
+
+    if recipients.split(/\s+/).count > max_chans
+      errors.add(:recipients, "are limited to #{max_chans}")
+    end
+  end
+
+  def max_chans
+    begin
+      max = Gitlab.config.irker['max_channels']
```

Comment:
It's the maximum number of recipients allowed per project, in order not to flood too many channels.